bool    relay_present
bool[]  relay_on
